app/merger.go

```diff
@@ -53,9 +53,10 @@ func (smartMerger) Merge(reports []report.Report) report.Report {
 		c <- r
 	}
 	for ; l > 1; l-- {
-		go func(left, right report.Report) {
+		left, right := <-c, <-c
+		go func() {
 			c <- left.Merge(right)
-		}(<-c, <-c)
+		}()
```

Comment:
It's a fair point that it could've been a seperate PR as long as it was merged first - this was just less accounting.